.gitlab-ci.yml

```diff
@@ -0,0 +1,92 @@
+workflow:
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+      auto_cancel:
+        on_new_commit: interruptible
+    - if: $CI_COMMIT_BRANCH == "main"
+      auto_cancel:
+        on_new_commit: conservative
+    - if: $CI_COMMIT_TAG
+      auto_cancel:
+        on_new_commit: conservative
+
+default:
+  interruptible: true
+  before_script:
+    # FIXME: build minimal el9 container, or use container-forge
+    - dnf install -y java-21-openjdk maven wget tree ruby python3-pip
+
+stages:
+  - build-stage
+  - test-stage
+  - deploy-stage
+
+build:
+  stage: build-stage
+  script:
+    - ./build-coatjava.sh -T4 --unittests --spotbugs --quiet --no-progress
+    - tar -czf coatjava.tar.gz coatjava
+  artifacts:
+    when: always
+    expire_in: 1 day
+    paths:
+      - coatjava
+
+depana:
+  allow_failure: true
+  stage: test-stage
+  dependencies: [build]
+  script:
+    - mvn dependency:tree -Ddetail=true --no-transfer-progress
+    - mvn dependency:analyze -DfailOnWarning=true -pl '!org.jlab.coat:coat-libs' --no-transfer-progress
+
+eb:
+  stage: test-stage
+  dependencies: [build]
+  script:
+    - cd validation/advanced-tests && ./run-eb-tests.sh -100 ${ARG}
+  parallel:
+    matrix:
+      - ARG: electronproton 
+      - ARG: electronprotonC 
+      - ARG: electrongamma 
+      - ARG: electronneutronC
+      - ARG: electronFTpion
+
+jacoco:
+  allow_failure: true
+  stage: test-stage
+  needs: [build]
```

Comment:
YAGNI (at least not yet):
```suggestion
```
You could also probably remove the `dependencies:` in most jobs too, since if you don't have `dependencies:`, all artifacts from the previous stage are supposed to be available in the current stage. Up to you.

Reply:
Actually, this one it really explicitly wanted, even though it seemed to me too that it shouldn't .  We can fix it after the move to gitlab.